### DISCUSSION TITLE
Is there any initiative for a DefinitelyTyped @types/chakra-ui library?

### DISCUSSION THREAD

**Discussion by jdpaterson** [chakra-ui/chakra-ui#3277] [Ideas] [answered]
This would suit my needs nicely. Currently having trouble adding chakra-ui types to a global scope. 

What I would like to be able to do is have access to the chakra types without having to import them. Something like this: 

```
import { Box } from '@chakra-ui/react'

interface CustomBox extends BoxProps {
  abc: string
}

but currently it cannot find BoxProps unless I import it directly. 

I can get this to work with other libraries like googlemaps.

**Comment by chasinhues** [comment, marked as answer]
Not to my knowledge. Chakra is built with TS and exports its own types so they can be used. 

I might be missing something, but why can't you just import `BoxProps` from the Chakra package?

**Reply by jdpaterson** [reply]
It's because I put my types in the global scope using .d.ts files and triple-slash directives. So if I have an interface that wants to extend BoxProps then I have to find a way to import BoxProps in my .d.ts file, but I'm not finding any way to do that successfully. Maybe I just need to re-evaluate my implementation. But again with something like GoogleMaps it works, I think because googlemaps is DefinitelyTyped
